src/System.Linq/src/System/Linq/Zip.cs

```diff
@@ -28,6 +28,9 @@ public static partial class Enumerable
             return ZipIterator(first, second, resultSelector);
         }
 
+        public static IEnumerable<(TFirst, TSecond)> Zip<TFirst, TSecond>(this IEnumerable<TFirst> first, IEnumerable<TSecond> second)
+            => first.Zip(second, (x, y) => (x, y));
```

Comment:
Should we avoid the delegate invocation on every iteration by having the implementation instead be expanded out like the original overload is? e.g.
```C#
public static IEnumerable<(TFirst, TSecond)> Zip<TFirst, TSecond>(this IEnumerable<TFirst> first, IEnumerable<TSecond> second)
{
    if (first == null || second == null)
    {
        throw Error.ArgumentNull(first == null ? nameof(first) : nameof(second));
    }

    return ZipIterator(first, second, resultSelector);
}
 
private static IEnumerable<(TFirst, TSecond)> ZipIterator<TFirst, TSecond>(IEnumerable<TFirst> first, IEnumerable<TSecond> second)
{
    using (IEnumerator<TFirst> e1 = first.GetEnumerator())
    using (IEnumerator<TSecond> e2 = second.GetEnumerator())
    {
        while (e1.MoveNext() && e2.MoveNext())
        {
            yield return (e1.Current, e2.Current);
        }
    }
}
```